src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -399,6 +463,143 @@ public Configuration getHadoopConfiguration() {
     return map;
   }
 
+  /**
+   * Performs all substitutions on `logs.dir` and calculates the final directory for a topic
+   *
+   * @param topic - String - the topic whose directory to find
+   * @return String - the directory name and path
+   */
+  public String getLogsDirFromTopic(String topic) {
+    return getDirFromTopic(getString(LOGS_DIR_CONFIG), topic, logDirGroupsMaxIndex);
```

Comment:
Have you verified that you've replaced all of the call-sites for `getString(LOGS_DIR_CONFIG)` and `getString(TOPICS_DIR_CONFIG)` with your new function?

Reply:
yes